cmd/crypto/kes.go

```diff
@@ -229,25 +242,55 @@ func (c *kesClient) GenerateDataKey(name string, context []byte) ([]byte, []byte
 	}
 
 	url := fmt.Sprintf("%s/v1/key/generate/%s", c.addr, url.PathEscape(name))
-	resp, err := c.httpClient.Post(url, "application/json", bytes.NewReader(body))
+
+	const limit = 1 << 20 // A plaintext/ciphertext key pair will never be larger than 1 MB
+	resp, err := c.postRetry(url, bytes.NewReader(body), limit)
 	if err != nil {
 		return nil, nil, err
 	}
+
+	return resp.Plaintext, resp.Ciphertext, nil
+}
+
+func (c *kesClient) post(url string, body io.Reader, limit int64) (*Response, error) {
+	resp, err := c.httpClient.Post(url, "application/json", body)
+	if err != nil {
+		return nil, err
+	}
 	if resp.StatusCode != http.StatusOK {
-		return nil, nil, c.parseErrorResponse(resp)
+		return nil, c.parseErrorResponse(resp)
 	}
-	defer resp.Body.Close()
+	defer xhttp.DrainBody(resp.Body)
```

Comment:
We must not arbitrary amounts of data from any response body. This must be limited beforehand.

Reply:
Are you assuming KES servers are adversaries :D

Reply:
You may not talk to an authentic KES server. In general, you cannot go ahead and just assume that the server behaves properly.
I think draining the body is not necessary / useful at all. If you get a response body and consume it - then closing the body should be sufficient - or am I missing something?

Otherwise, all code out there just using `resp.Body.Close()` would be broken and it would be `net/http`'s resp. to fix it.

Reply:
The issue is Go doesn't reuse it @aead until you hit EOF this is the known problem

Reply:
https://github.com/golang/go/issues/26095#issuecomment-562521360

Reply:
It is to ensure that body is ready to completion, as we are never sure how much of json decoder would read. It is not expected to read till EOF, neither does io.Limit reader. So essentially you are not draining ant last remaining data on the wire.

This simply takes care of all those situations that may be already present.

Reply:
This is not the case here. This only applies to HTTP/1.1 connections with TCP keep alive. Draining a response body in case of HTTP/2 seems not necessary.

```
// The default HTTP client's Transport may not
// reuse HTTP/1.x "keep-alive" TCP connections if the Body is
 // not read to completion and closed.
```